### DISCUSSION TITLE
Two "Projects" Locally into One Project on Github

### DISCUSSION THREAD

**Discussion by SushiKishi** [community/community#68843] [Programming Help]
It was a little bit difficult, especially since I'm using Visual Studio 2017 so I had to find older tutorials long buried, but I managed to push something to Github! I'm using VS2017 for C# to create a plugin for LiveSplit, but the main thing is going to be a web app in PHP/JavaScript. I'm using Komodo IDE for that.

All of which is fine; I know GitHub is supportive for collaborative efforts so the fact it's two different IDEs isn't really an issue any more than two users pushing updates would be an issue. Where I'm confused is how to push the two different 'halves' of my project to the same GitHub. Visual Studio 2017 has all of my LiveSplit-related code and files in the default Windows repo folder (C:\users\whatever\repos), but I have my Komodo project working out of where I have XAMPP set up. So each IDE has only the files for its "half" of the project. I'm not sure what would happen if I push them as two separate halves. I'd just try it and find out, but when I try to push from Komodo I'm getting an Error 1 for git.exe in the SCC Push window. I don't know if it's because it sees there's wildly different stuff in the repository and it wants me to pull first or what. But if I pull, then I'll have all my VS2017 files in my Komodo project directory...

I'm just not sure how to properly put them all together. Part of my concern is that in order to get my LiveSplit plugin to build, I first had to clone LiveSplit, build it, then make the new project and import etc. etc -- the point is that on my local disc in my project's repo, there's a lot of files that don't actually get pushed because they're used to build LiveSplit, and (for the most part*) only the files I'm making are getting pushed to GitHub for my project -- certainly not the entire LiveSplit repository. So if I end up mix-matching my projects, I don't want Komodo to grab allllll the LiveSplit files I cloned then pushing those like they were *my* project. But I don't even know if that makes sense because honestly Git-ing and GitHub-ing is still a little bit over my head...oi!

Can someone help clear it up a little bit for me?

**Comment by ghost** [comment]
If you don't want certain files or folders to be pushed to the repository, create a `.gitignore` file. In there, type the paths of your files or folders that you don't want to be pushed. Not sure if that's helpful to you but hey, worth a shot :+1:

**Comment by sreshtech** [comment]
I can definitely help clear up some of the confusion around how to structure your project with Git and GitHub for your situation. It sounds like you have two different parts of your project developed in two different IDEs (Visual Studio for LiveSplit and Komodo IDE for the web app) and you want to keep them in the same GitHub repository without mixing them up.???

Here's a general approach you can follow:

1. Initialize a Git Repository in Your Web App Directory: Open your Komodo IDE project directory (where your PHP/JavaScript code is) in your terminal or Git GUI tool, and initialize a Git repository 

2. Create a `.gitignore` File-In this repository, create a `.gitignore` file to specify which files and directories should be ignored by Git. You can add patterns to this file to exclude build files, temporary files, and any other files you don't want to include in your repository. For example:
   #=> Ignore build and temporary files
   /build/
   /temp/
   ```

3. Commit and Push Your Web App Code: Commit and push your web app code to this repository. You can do this using Git commands or your Git GUI tool. This will create a repository for your web app on GitHub.

4. Initialize Another Git Repository in Your LiveSplit Directory: Now, navigate to your LiveSplit directory (where Visual Studio is), and initialize another Git repository there 

5. Commit and Push Your LiveSplit Code Separately: Commit and push only the relevant code and files for your LiveSplit plugin to this repository. Make sure to ignore the LiveSplit build files and any other unnecessary files.

By following this approach, you'll have two separate Git repositories—one for your web app and one for your LiveSplit plugin—in the same GitHub repository. They won't interfere with each other, and you can work on them independently. When someone clones the GitHub repository, they can choose to clone both repositories or just the one they are interested in.

Remember to commit and push regularly to keep your GitHub repositories up to date. Additionally, consider using branching and pull requests to manage changes and contributions from collaborators effectively.
